<chrono>: Allow the parsing of year + day-of-the-year to a time_point using from_stream

I'm not able to parse a string representing a date in the format `<year>.<day-of-the-year>.<hours>.<minutes>.<seconds>` using `std::from_stream` to a `std::chrono::time_point`. The following code fails for me on Visual Studio 2019 v16.10.

```
auto toTimePoint(const std::string& val)
{
    std::istringstream ss(val);
    std::chrono::system_clock::time_point tp;
    std::chrono::from_stream(ss, "%Y.%j.%H.%M.%S", tp)
    return tp;
}
```

Digging into the implementation, I've found that the code fails on [this line](https://github.com/microsoft/STL/blob/3cafa97eecdbfde41ea5c09126f877a7eb97f9e9/stl/inc/chrono#L4083) of the `chrono` header, because the `_Used` bit mask misses both the `_F_day` and `_F_month` flags: in this case this requirement should be fulfilled by the presence of the `_F_doy` flag, shouldn't it?

Also check this [question](https://stackoverflow.com/questions/67726168/how-to-parse-2021-147-17-33-45-using-howard-hinnants-date-library-or-the-equ?noredirect=1#comment119717978_67726168) on SO.

The above fix fixes std::chrono::from_stream but std::chrono::parse is still broken - 
`    auto toTimePoint(const std::string& val)
    {
        std::chrono::system_clock::time_point tp = std::chrono::parse<std::string::value_type, std::string::traits_type, std::string::allocator_type, std::string>("%Y.%j.%H.%M.%S", val);
        return tp;
    }`

fails to deduce the fifth unnamed template argument, which i think is using sfinae to select the correct from_stream to use?
`
error C2672: 'std::chrono::parse': no matching overloaded function found
error C2783: 'auto std::chrono::parse(const std::basic_string<_Elem,_Traits,_Alloc> &,_Parsable &)': could not deduce template argument for '<unnamed-symbol>'
`

> 
> 
> The above fix fixes std::chrono::from_stream but std::chrono::parse is still broken -
> 
> ```
>     auto toTimePoint(const std::string& val)
>     {
>         std::chrono::system_clock::time_point tp = std::chrono::parse<std::string::value_type, std::string::traits_type, std::string::allocator_type, std::string>("%Y.%j.%H.%M.%S", val);
>         return tp;
>     }
> ```
> 
> fails to deduce the fifth unnamed template argument, which i think is using sfinae to select the correct from_stream to use?
> 
> ```
> error C2672: 'std::chrono::parse': no matching overloaded function found
> error C2783: 'auto std::chrono::parse(const std::basic_string<_Elem,_Traits,_Alloc> &,_Parsable &)': could not deduce template argument for '<unnamed-symbol>'
> ```

IIUC This is not the correct way to use `parse`. The intended usage is `std::istringstream(val) >> std::chrono::parse("%Y.%j.%H.%M.%S"s, tp)`

--- a/stl/inc/chrono
+++ b/stl/inc/chrono
@@ -4122,12 +4122,9 @@ namespace chrono {
 
             if constexpr (_Can_rep_day) {
                 if (_For_time_point) {
-                    year_month_day _Ymd;
-                    if (_Make_year_month_day(_Ymd, true)) {
-                        _Result += _CHRONO duration_cast<_DurationType>(static_cast<sys_days>(_Ymd).time_since_epoch());
-                    } else {
-                        return false;
-                    }
+                    const year_month_day _Ymd{year{*_Century + *_Two_dig_year},
+                        month{static_cast<unsigned int>(*_Month)}, day{static_cast<unsigned int>(*_Day)}};
+                    _Result += _CHRONO duration_cast<_DurationType>(static_cast<sys_days>(_Ymd).time_since_epoch());
                 } else if (_Used & _F_doy) {
                     _Result += _CHRONO duration_cast<_DurationType>(days{*_Day_of_year});
                 }
@@ -4157,7 +4154,7 @@ namespace chrono {
         template <class _DurationType>
         _NODISCARD bool _Make_time_point(_DurationType& _Dur, _LeapSecondRep _Leap) {
 
-            const bool _Consistent{_Calculate_hour24() && _Calculate_year_fields()};
+            const bool _Consistent{_Calculate_hour24() && _Calculate_year_fields() && _Calculate_ymd()};
             if (!_Consistent || !_Apply_duration_fields<_Parse_tp_or_duration::_Time_point>(_Dur)) {
                 return false;
             }

--- a/tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp
+++ b/tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp
@@ -718,6 +718,8 @@ void parse_calendar_types_basic() {
     test_parse("366 2004-12-31", "%j %F", ymd);
     assert(ymd == 2004y / December / last);
 
+    test_parse("82 1882", "%j %Y", ymd);
+    assert(ymd == 23d / March / 1882y);
     fail_parse("366 2001", "%j %Y", ymd);
     fail_parse("367 2004", "%j %Y", ymd);
 
@@ -1086,6 +1088,10 @@ void parse_timepoints() {
     assert(st == sys_days{23d / June / 1912y});
     test_parse("1912-06-23 23:59:59", "%F %T", st);
     assert(st == sys_days{23d / June / 1912y} + 23h + 59min + 59s);
+
+    // GH-1938: parsing timepoint with year and day-of-year
+    test_parse("1882.82", "%Y.%j", st);
+    assert(st == sys_days{23d / March / 1882y});
 }
 
 void parse_wchar() {

